core/src/main/java/org/apache/james/core/quota/QuotaLimitKey.java

```diff
@@ -0,0 +1,84 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+
+import com.google.common.base.MoreObjects;
+
+public class QuotaLimitKey {
```

Comment:
Ok we gonna have to do the choice of doing this one way and not two: https://github.com/apache/james-project/pull/1813

@quantranhong1999 adds the key as a subclass of QuotaCurrentValue, here it's a separate class. Personally I think this could be a subclass of QuotaLimit?